### ISSUE TITLE
Dask task's displaying node names

### ISSUE THREAD

**Issue by janhurst** [DAGWorks-Inc/hamilton#161]
**Is your feature request related to a problem? Please describe.**
Is it possible to have the Dask status/dashboards report the names of the functions rather than a generic `node_fn` label? I'm not sure if this is a Dask concern or something that can be manipulated within Hamilton?

Below, a pipeline is executing on a Dask worker, however I can't easily/obviously understand what part of the DAG is being executed.
![image](https://user-images.githubusercontent.com/16224889/236233993-9e2e853d-e45d-47c5-a89c-cda4bb90eb00.png)

**Describe the solution you'd like**
The name of the function defined in the DAG (which I think is in some part related to the node name?) appears in the Dask dashboards rather than `node_fn`

**Additional context**
I am trying to use Hamilton to define a big bunch of related workflows, farm them out to a Dask scheduler and use Dask's dashboards as a proxy task monitor.

**Comment by elijahbenizzy** [comment]
OK, I think this should be easy. Just to check, you're using the Dask graph adapter?

I'm thinking all we add is the `name=node.name` [here](https://github.com/DAGWorks-Inc/hamilton/blob/main/hamilton/experimental/h_dask.py#L116). Pretty sure its the `name` and not hte `dask_key_name` although not 100% sure. Will have to mess with it a bit.

**Comment by elijahbenizzy** [comment]
OK, we'll need to sit down and run some tests, but mind giving it a try with this: https://github.com/DAGWorks-Inc/hamilton/pull/163? Should be as easy as:

`pip install git+https://github.com/dagworks-inc/hamilton.git@dask-improvement`

**Comment by skrawcz** [comment]
@janhurst yeah let us know if this version solves your issue -- we'll then merge and deploy it :)

**Comment by janhurst** [comment]
> OK, I think this should be easy. Just to check, you're using the Dask graph adapter?

Yes, the Dask graph adapter

> OK, we'll need to sit down and run some tests, but mind giving it a try with this: #163? Should be as easy as:
> 
> `pip install git+https://github.com/dagworks-inc/hamilton.git@dask-improvement`

No luck... I'm moderately sure that my local venv and the dask scheduler/workers are running off the tag :)

![image](https://user-images.githubusercontent.com/16224889/236362218-bfcedaed-1834-4351-b984-296a8be270b4.png)

![image](https://user-images.githubusercontent.com/16224889/236362092-33a90209-2533-47a3-b7a3-32651f482748.png)

**Comment by skrawcz** [comment]
@janhurst  :/ mind putting in details of your setup so I can try to recreate it?

**Comment by janhurst** [comment]
I am using docker compose to bring up a scheduler and workers. The Dockerfile I am using is roughly:
```
FROM ghcr.io/dask/dask:latest-py3.9

RUN pip install \
    git+https://github.com/dagworks-inc/hamilton.git@dask-improvement \
    "pandas>=1.3.5,<1.4" \
    "numpy>=1.23,<1.24" 

ENV PYTHONPATH=/opt/app/src
COPY src /opt/app/src   
```

My local `.venv` is using the same pandas and numpy pins along with `dask[complete]>=2023.4`. There is quite a few other packages in my local `.venv`, usual suspects of dev tools (eg pytest etc), and a bunch of jupyter packages etc.
Sorry about the pandas and numpy pins, its a frustrating constraint coming from Dataiku.

The hamilton pipeline is a fan out to sub dags and then collect back, similar to our discussion elsewhere. The nodes in the sub DAG essentially call out to a slow/expensive database and serialize to parquet.

**Comment by elijahbenizzy** [comment]
OK, I've managed to reproduce it. This is hacky but I *hope* should get you unblocked -- there be some dragons around dask. If you try repulling the latest version from that it should work (not exactly sure how to do that with pip but I think if you uninstall it'll work. 

Currently I appended a UUID cause dask was giving me trouble with the raw node name...

**Comment by elijahbenizzy** [comment]
For future reference, the easiest way to test this on Hamilton is:
1. Run a dask pipeline locally (`examples/dask/hello_world`, for example)
2. Add a *very* long sleep statement in a function
3. Run it

Then go to http://localhost:8787/status and you can see the dashboard.

**Comment by janhurst** [comment]
thanks! seeing node names now!

I'll have to experiment with my DAG/sub-DAG structure... i have setup the DAG so that i have many sub DAGs, when in fact there are 6 groups of sub dags.... being able to see those groups in a Dask dashboard would be great too, but I think the fanout/collect stuff we have been talking about needs to settle a bit before i worry about it!

![image](https://user-images.githubusercontent.com/16224889/236379560-601b975f-3b1d-4a85-919e-0ed3068bd999.png)

**Comment by elijahbenizzy** [comment]
Excellent! Those node names make about as much sense to me as not having them but I think they make more sense to you 😆 

And I think we can do that if dash allows, we have all the metadata.

**Comment by skrawcz** [comment]
@janhurst can we close this issue? or?

**Comment by elijahbenizzy** [comment]
@skrawcz we should merge first, no?

**Comment by janhurst** [comment]
I think we've addressed the original intent raised here.. I can see much more meaningful node info in dask now!

there might be some more we could do but perhaps better served in a new issue/feature request?
